core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/azure/AzureMetadataSetup.java

```diff
@@ -131,14 +132,22 @@ public InstanceSyncState getState(Stack stack, String instanceId) {
         Map<String, String> vmContext = createVMContext(instanceId);
         AzureCredential credential = (AzureCredential) stack.getCredential();
         AzureClient azureClient = azureStackUtil.createAzureClient(credential);
-        InstanceSyncState instanceSyncState = IN_PROGRESS;
+        InstanceSyncState instanceSyncState;
         try {
-            if ("Running".equals(azureClient.getVirtualMachineState(vmContext))) {
+            String virtualMachineState = azureClient.getVirtualMachineState(vmContext);
+            if ("Running".equals(virtualMachineState)) {
                 instanceSyncState = RUNNING;
-            } else if ("Suspended".equals(azureClient.getVirtualMachineState(vmContext))) {
+            } else if ("Suspended".equals(virtualMachineState)) {
                 instanceSyncState = STOPPED;
+            } else if (isActualStatusDesired(virtualMachineState, "RunningTransitioning", "SuspendedTransitioning",
+                    "Starting", "Suspending", "Deploying", "Deleting")) {
+                instanceSyncState = IN_PROGRESS;
+            } else {
+                instanceSyncState = UNKNOWN;
             }
         } catch (Exception ex) {
+            String msg = String.format("Failed to retrieve status of instance(%s) from Azure, set InstanceSyncState to 'DELETED'.", instanceId);
+            LOGGER.info(msg, ex);
             instanceSyncState = DELETED;
```

Comment:
I think the exception should be parsed if it is a "not found" or something else. If it's a not found, the returned state can be deleted, otherwise (e.g.: connection issue) mark it as unknown because we don't want to delete it.